content/components/segmented-control.mdx

```diff
@@ -0,0 +1,426 @@
+    width="456"
+  />
+</Box>
+
+### Adapting to narrow areas
+
+There are cases where a segmented control needs to be placed in a spot that is too narrow to accommodate all of its options. In these cases, the control can be adapted to fit in the narrow area without sacrificing usability.
+
+<Box
+  as="figure"
+  display="flex"
+  alignItems="center"
+  flexDirection={['column', 'column', 'column', 'column', 'row']}
+  m={0}
+  mb={3}
+  sx={{gap: 3}}
+>
+  <figcaption>
+    <Text as="h4" fontSize={2} fontWeight="bold" m={0}>
+      Convert to an action menu
+    </Text>
+    <Text as="p" mt="0">
+      Swapping out the segmented control for an action menu is especially helpful when:
+      <Box as="ul" mt={1} pl="1rem">
+        <li>The icons are cannot be understood without visible text labels</li>
+        <li>The control is stil too wide when text labels are hidden</li>
+      </Box>
+    </Text>
+  </figcaption>
+  <img
+    src="https://user-images.githubusercontent.com/2313998/180624846-a906563a-0878-44ce-89d1-46905b190d20.png"
+    role="presentation"
+    width="456"
+  />
+</Box>
+
+<Box
+  as="figure"
+  display="flex"
+  alignItems="center"
+  flexDirection={['column', 'column', 'column', 'column', 'row']}
+  m={0}
+  mb={5}
+  sx={{gap: 3}}
+>
+  <figcaption>
+    <Text as="h4" fontSize={2} fontWeight="bold" m={0}>
+      Only show icons
+    </Text>
+    <Text as="p" mt="0">
+      If your segmented control uses icons that can be understood without a visible text label, then you can hide the
+      text labels and rely only on icons. The text from the hidden label will still be visible in a tooltip.
+    </Text>
+  </figcaption>
+  <img
+    src="https://user-images.githubusercontent.com/2313998/180624846-a906563a-0878-44ce-89d1-46905b190d20.png"
+    role="presentation"
+    width="456"
+  />
+</Box>
+
+<DoDontContainer>
+  <Do>
+    <img
+      src="https://user-images.githubusercontent.com/2313998/180624845-49eb0b79-21ee-43d0-b4c8-8e8528359427.png"
+      role="presentation"
+    />
+    <Caption>
+      Replace the segmented control with an action menu or use icons that are easily understood without text labels.
+    </Caption>
+  </Do>
+  <Dont>
+    <img
+      src="https://user-images.githubusercontent.com/2313998/180624844-c81c94b4-a004-4c89-8aea-2470c429dd75.png"
+      role="presentation"
+    />
+    <Caption>Wrap the buttons to multiple lines, use an overflow button, or remove buttons</Caption>
+  </Dont>
+</DoDontContainer>
+
+## Accessibility
+
+### Keyboard navigation
+
+<Box
+  as="figure"
+  display="flex"
+  alignItems="flex-start"
+  flexDirection={['column-reverse', 'column-reverse', 'column-reverse', 'column-reverse', 'row']}
+  m={0}
+  mb={3}
+  sx={{gap: 3}}
+>
+<div>
+
+A segmented control is treated like a [toolbar](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/toolbar_role), not a [radio group](https://developer.mozilla.org/en-US/docs/web/accessibility/aria/roles/radiogroup_role).
```

Comment:
What's the recommendation when a SegmentedControl is used as a tab (akin to TabNav/UnderlineNav behavior)?

Reply:
I'm bringing this to a11y office hours today: https://github.com/github/accessibility/issues/1556#issuecomment-1204047337

I'll keep you posted.

Reply:
The recommendation was to not use a segmented control like a tab: https://github.rewatch.com/video/jlbodxdruw5bzc8b-accessibility-office-hours-august-3-2022?t=475.87

cc @simurai

Reply:
@mperrotti Thanks for bringing this up in the a11y office hour.

> The recommendation was to not use a segmented control like a tab

As I understood it, `SegmentedControl` should behave like a list of items where each `tab` keypress moves to the next item and you can make it active by pressing `enter` or `space`.

But switching between "Write/Preview" in the comment box should use what they call a "composite widget". Where only one tab-stop exists and then you would use the arrow keys to toggle between "Write" and "Preview". Basically keep the current [TabNav](https://primer.style/view-components/components/alpha/tabnav). Keep using tabs should also make it more clear that the two options are mutually exclusive.